### PR TITLE
fix table formatting in container

### PR DESCRIPTION
**Pull request by wilzbach**
follow-up to #4354 - you currently have to scroll _a lot_ to reach the container table.

http://dlang.org/phobos/std_container.html

I took the opportunity to reformat the table properly.

### CODE REVIEW

**Comment by aG0aep6G** [comment]
As with #4358, dmd's handling of empty lines should be fixed instead of working around it here.

**Comment by wilzbach** [comment]
> As with #4358, dmd's handling of empty lines should be fixed instead of working around it here.

I agree, but I don't know how long this will take us & currently the docs look ugly / broken.
Btw this PR also cleans up the container table and there's still enough broken stuff to test if we want to fix ddoc-

http://dtest.thecybershadow.net/artifact/website-fc52239a89c0f31ce8fab3844f156b775682a9e6-4a2b40a9d2e96f8be33a50e040555c47/web/phobos-prerelease/std_container.html
https://dlang.org/phobos/std_container.html

**Comment by DmitryOlshansky** [comment]
LGTM it's worthwhile to fix docs if we can without waiting on DMD side

**Comment by wilzbach** [comment]
Anyone else in favor of cleaning up the docs and  removing  those two pages of whitespace visible to the reader? 
(this pr also reformats the table in a more human readable way)

File: std/container/package.d [hunk 1]
```diff
@@ -215,189 +208,287 @@ value $(D x), which could be a single element (in which case $(D n$(SUBSCRIPT x)
...
+    $(TD $(D c.length))
+    $(TDNW $(D log n$(SUBSCRIPT c)))
+    $(TD Returns the number of elements in the _container.)
+)
+$(TR
+    $(TDNW $(D c.length = n))
+    $(TDNW $(D n$(SUBSCRIPT c) + n))
+    $(TD Forces the number of elements in the _container to $(D n).
+        If the _container ends up growing, the added elements are initialized
+        in a_container-dependent manner (usually with $(D T.init)).)
```
Comment (aG0aep6G) [inline std/container/package.d]: A space went missing here. "a_container" should be "a _container".

**Comment by wilzbach** [comment]
> A space went missing here. "a_container" should be "a _container".

rebased.

**Comment by aG0aep6G** [comment]
Ok, the table changes look good to me. These are responsible for the huge white void before the table.

Regarding the other blank lines, I'm still not happy that we have to workaround Ddoc's weirdness like that. We'd have to go over all of the documentation and do this, making it less readable in source.

But maybe this is necessary. I don't see an obvious way out. Looks good to me then.

Also, I think you have missed some blank lines. Or are these aesthetic choices?

* after "Note that make [...]"
* after "This is in fact [...]"
* after "If the documentation [...]"
* before and after "When any range [...]"
* after "The following table [...]"

**Comment by aG0aep6G** [comment]
> But maybe this is necessary. I don't see an obvious way out.

I came up with something: https://github.com/dlang/dlang.org/pull/1317

**Comment by wilzbach** [comment]
> Also, I think you have missed some blank lines. Or are these aesthetic choices?

No that's (as you probably know by know) still an artificat from DDoc.

> Regarding the other blank lines, I'm still not happy that we have to workaround Ddoc's weirdness like that. 

I agree, let's fix Ddoc! (removed them from this PR)

**Comment by wilzbach** [comment]
> I agree, let's fix Ddoc! (removed them from this PR)

I created an issue for this and closed the other PRs

https://issues.dlang.org/show_bug.cgi?id=16110

As the current container page has this huge whitespace offset, I would argue it's worth the noise as we don't know how long it takes to fix ddoc.

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**